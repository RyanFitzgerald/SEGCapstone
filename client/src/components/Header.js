import React from 'react';
import {Link} from 'react-router-dom';
import Logo from '../logo.png';
import ClickOutside from './ClickOutside';

function Header(props) {
  let menu, userMenu, username, level;
  let sessionUser = JSON.parse(localStorage.getItem('user'));
  if (sessionUser) {
    username = sessionUser.name;
    level = sessionUser.role.level;
  } else {
    username = '';
    level = 0;
  }

  function handleMenuToggle() {
    if (menu.classList.contains('active')) {
      menu.classList.remove('active');
    } else {
      menu.classList.add('active');
    }
  }

  function handleUserMenuToggle() {
    if (userMenu.classList.contains('active')) {
      userMenu.classList.remove('active');
    } else {
      userMenu.classList.add('active');
    }
  }

  function closeMenu() {
    if (userMenu.classList.contains('active')) {
      userMenu.classList.remove('active');
    }
  }

  return (
    <div className="header clearfix">
      <div className="header__logo">
        <Link to="/"><img src={Logo} alt="Renovaction" /></Link>
      </div>

      <div className="header__user">
        <span className="header__usertoggle" onClick={handleUserMenuToggle}>Hello, {username} <i className="fa fa-chevron-down" aria-hidden="true"></i></span>

        <ClickOutside clickOutsideEvent={closeMenu}>
          <ul ref={ele => userMenu = ele} onClick={closeMenu}>
              <li><Link to='/account'>Account Settings</Link></li>
              <li onClick={props.logout}><Link to="/">Logout</Link></li>
          </ul>
        </ClickOutside>

        <span className="header__menutoggle" onClick={handleMenuToggle}>
            <i className="fa fa-bars" aria-hidden="true"></i>
        </span>
      </div>

      <div className="header__nav" ref={ele => menu = ele}>
        <ul>
          <li><Link className={(props.activeTab === 1) ? 'header__link header__link--active' : 'header__link'} to="/">Home</Link></li>
          <li><Link className={(props.activeTab === 2) ? 'header__link header__link--active' : 'header__link'} to="/projects">Projects</Link></li>
          <li><Link className={(props.activeTab === 3) ? 'header__link header__link--active' : 'header__link'} to="/clients">Clients</Link></li>
          {level >= 2 &&
          <li><Link className={(props.activeTab === 4) ? 'header__link header__link--active' : 'header__link'} to="/stats">Statistics</Link></li>
          }
          {level >= 2 &&
          <li><Link className={(props.activeTab === 5) ? 'header__link header__link--active' : 'header__link'} to="/settings/users">Settings</Link></li>
          }
        </ul>
      </div>
    </div>
  );
};

export default Header;