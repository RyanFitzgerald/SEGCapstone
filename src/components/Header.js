import React, {Component, PropTypes} from 'react';

class Header extends Component {
    render() {
        return (
            <div id="header" className="clearfix">
                <div id="header-page-title">
                    <a href="#" id="sidebar-toggle"><i className="fa fa-bars" aria-hidden="true"></i></a>
                    <h1>{this.props.title}</h1>
                </div>
                {/* <!-- End header-page-title --> */}
                <div id="header-user-actions">
                    <ul>
                        <li className="hide-for-small-only">
                            <a href="#">Help Center</a>
                        </li>
                        <li className="hide-for-small-only">
                            <a href="#">Notifications</a>
                        </li>
                        <li id="user-select">
                            <a href="#">Ryan Fitzgerald <i className="fa fa-chevron-down" aria-hidden="true"></i></a>
                            <ul className="rounded shadow-medium" id="user-dropdown">
                                <li><a href="#" className="show-for-small-only">Help Center</a></li>
                                <li><a href="#" className="show-for-small-only">Notifications</a></li>
                                <li><a href="#">Settings</a></li>
                                <li><a href="#">Sign Out</a></li>
                            </ul>
                        </li>
                    </ul>
                </div>
                {/* <!-- End header-user --> */}
            </div>
        );
    }
}

Header.propTypes = {
    title: PropTypes.string.isRequired
};

export default Header;
