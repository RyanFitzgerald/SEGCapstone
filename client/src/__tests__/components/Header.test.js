import React from 'react';
import Header from '../../Components/Header';

describe('<Header />', () => {
  beforeAll(() => {
    // Add necessary info to user
    const user = {
      name: 'John',
      role: {
        level: 1
      }
    };
    sessionStorage.setItem('user', JSON.stringify(user));
  });

  it('should render correctly', () => {
    // Shallow render it
    const wrapper = shallow(
      <Header/>
    );

    // Compare to old snapshot
    expect(shallowToJson(wrapper)).toMatchSnapshot();

    // Ensure username is in header
    expect(wrapper.find('.header__usertoggle').prop('children')).toContain('John');
  });
});
