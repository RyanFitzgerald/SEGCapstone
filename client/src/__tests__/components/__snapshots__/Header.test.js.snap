// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render correctly 1`] = `
<div
  className="header clearfix"
>
  <div
    className="header__logo"
  >
    <Link
      replace={false}
      to="/"
    >
      <img
        alt="Renovaction"
        src="logo.png"
      />
    </Link>
  </div>
  <div
    className="header__user"
  >
    <span
      className="header__usertoggle"
      onClick={[Function]}
    >
      Hello, 
      John
       
      <i
        aria-hidden="true"
        className="fa fa-chevron-down"
      />
    </span>
    <ul>
      <li>
        <Link
          replace={false}
          to="/account"
        >
          Account Settings
        </Link>
      </li>
      <li>
        <Link
          replace={false}
          to="/"
        >
          Logout
        </Link>
      </li>
    </ul>
    <span
      className="header__menutoggle"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="fa fa-bars"
      />
    </span>
  </div>
  <div
    className="header__nav"
  >
    <ul>
      <li>
        <Link
          className="header__link"
          replace={false}
          to="/"
        >
          Home
        </Link>
      </li>
      <li>
        <Link
          className="header__link"
          replace={false}
          to="/projects"
        >
          Projects
        </Link>
      </li>
      <li>
        <Link
          className="header__link"
          replace={false}
          to="/clients"
        >
          Clients
        </Link>
      </li>
    </ul>
  </div>
</div>
`;
